### PR TITLE
Display text posts in the new campaign gallery.

### PR DESCRIPTION
**Pull request by DFurnes**
### What does this PR do?
This pull request adds support for text posts in Phoenix's new gallery component.

screen shot 2018-04-05 at 10 41 38 am

### Any background context you want to provide?
For now, "I beat bullying by..." is hard-coded. Let's think on how best to handle this in a follow-up PR!

### What are the relevant tickets/cards?
Closes [#155944555](https://www.pivotaltracker.com/story/show/155944555)!

### Checklist
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added appropriate feature/unit tests.
- [ ] Added screenshot to PR description of related front-end updates on **small** screens.
- [ ] Added screenshot to PR description of related front-end updates on **medium** screens.
- [ ] Added screenshot to PR description of related front-end updates on **large** screens.

### CODE REVIEW

**Comment by DFurnes** [comment]
@lkpttn I deviated a bit from the mocks: since people may have different length first names, I wasn't able to place the "tail" of the speech bubble in-line without it potentially overlapping. I also inherited the same "base" of the card as the photos use (so only the media portion changes) for consistency and a lil' cleaner implementation.

Give me a yell if either of these feel too "wrong" to you!

**Comment by DFurnes** [comment]
@weerd @mendelB This should be ready for review!

I added a screenshot, but feel free to take a peek at [the gallery](https://ds-phoenix-staging-pr-810.herokuapp.com/us/campaigns/mirror-messages/blocks/2iaAzD8xOsoUEiUMcokw4m) on the review app as well!

**Comment by DFurnes** [comment]
Added the alternating background colors in 137d430.

**Comment by weerd** [comment]
> For now, "I beat bullying by..." is hard-coded. Let's think on how best to handle this in a follow-up PR!

Oh ya. That's a good question. Lemme know if you have ideas? I could maybe toss the form label content from the Text Submission Action as a prop in the `details` JSON data I send to Rogue. If it's something that Rogue returns back to us, then that could be a potential solution. Not super keen on it, but it's something!

File: resources/assets/components/CampaignGalleryBlock/campaign-gallery.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
+@import '../../scss/next-toolbox';
+
+.campaign-gallery {
+  // Alternate adjacent 'chat-bubble' colors within a gallery.
```
Comment (weerd) [inline resources/assets/components/CampaignGalleryBlock/campaign-gallery.scss]: Noice :)

**Comment by weerd** [comment]
image

Small detail. Maybe use `bottom: -1px;` if possible, since there seems to sometimes be a slight sub-pixel rendering white background line that can show up. if you overlap it slightly it shouldn't show up :)

File: resources/assets/components/utilities/PostCard/PostCard.js [hunk 2]
```diff
@@ -24,15 +26,34 @@ export const postCardFragment = gql`
 
 const PostCard = ({ post, noun }) => {
   const reactionElement = <ReactionButton post={post} />;
+  let media = null;
+
+  // Render the appropriate media for this post:
+  switch (post.type) {
```
Comment (weerd) [inline resources/assets/components/utilities/PostCard/PostCard.js]: I dig this 💃

**Comment by DFurnes** [comment]
> Small detail. Maybe use `bottom: -1px;` on the callout triangle-thing if possible, since there seems to sometimes be a slight sub-pixel rendering white background line that can show up. if you overlap it slightly it shouldn't show up :)

Gah! Fixed in f0f18f0. 🔬

**Comment by DFurnes** [comment]
screen shot 2018-04-05 at 11 48 12 am

Scaled down the font-size at smaller widths in 8979563. Displayed above with a 120-character post.

**Pull request merged**